PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/actions/CreateFunctionAction.java

```diff
@@ -0,0 +1,214 @@
+import com.microsoft.intellij.util.AzureFunctionsUtils;
+import com.microsoft.intellij.util.PluginUtil;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.jps.model.java.JavaModuleSourceRootTypes;
+
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Writer;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class CreateFunctionAction extends CreateElementActionBase {
+    public CreateFunctionAction() {
+        super("Azure Function Class",
+                "newPage.dialog.prompt", IconLoader.getIcon(AzureFunctionSupportConfigurationType.ICON_PATH,
+                        AzureFunctionSupportConfigurationType.class));
+    }
+
+    @Override
+    protected PsiElement[] invokeDialog(Project project, PsiDirectory psiDirectory) {
+        PsiPackage pkg = JavaDirectoryService.getInstance().getPackage(psiDirectory);
+        // get existing package from current directoy
+        String hintPackageName = pkg == null ? "" : pkg.getQualifiedName();
+
+        CreateFunctionForm form = new CreateFunctionForm(project, hintPackageName);
+        List<PsiElement> psiElements = new ArrayList<>();
+        if (form.showAndGet()) {
+            Map<String, String> parameters = form.getTemplateParameters();
+            String triggerType = form.getTriggerType();
+            String packageName = parameters.get("packageName");
+            String className = parameters.get("className");
+
+            PsiDirectory directory = com.intellij.psi.util.ClassUtil.sourceRoot(psiDirectory);
+            String newName = packageName.replace('.', '/');
+            final FunctionTemplate bindingTemplate;
+            try {
+                bindingTemplate = form.getFunctionTemplate(triggerType);
+                final String functionClassContent = AzureFunctionsUtils.substituteParametersInTemplate(bindingTemplate, parameters);
+                Application application = ApplicationManager.getApplication();
+                application.runWriteAction(() -> {
+                    CreateFileAction.MkDirs mkDirs = ApplicationManager.getApplication().runWriteAction((Computable<CreateFileAction.MkDirs>) () -> new CreateFileAction.MkDirs(newName + '/' + className, directory));
+                    PsiFileFactory factory = PsiFileFactory.getInstance(project);
+                    CommandProcessor.getInstance().executeCommand(project, () -> {
+                        PsiFile psiFile = factory.createFileFromText(className
+                                + ".java", JavaFileType.INSTANCE, functionClassContent);
+
+                        psiElements.add(mkDirs.directory.add(psiFile));
+                    }, null, null);
+
+                    if (StringUtils.equalsIgnoreCase(triggerType, CreateFunctionForm.EVENT_HUB_TRIGGER)) {
```

Comment:
As we may add more triggers later, we could add a common workflow for local.settings later, get the new app settings from `CreateFunctionForm` and save them to local settings.

Reply:
done